controller/src/main.rs

```diff
@@ -62,6 +76,9 @@ async fn main() -> Result<()> {
         _ = controller_runner => {
             event!(Level::ERROR, "controller exited");
         },
+        _ = prometheus_server => {
```

Comment:
Super nit: the TODO on line 71 could now change to "// TODO if any of these fail..."